vulnerabilities/importers/pysec.py

```diff
@@ -0,0 +1,178 @@
+            raw_data["published"][0:19], "%Y-%m-%dT%H:%M:%S"
+        ).replace(tzinfo=timezone.utc)
+    else:
+        date_published = None
+        logger.error("date_published not found " + raw_data["id"])
+
+    severity = []
+    if "severity" in raw_data:
+        for sever_list in raw_data["severity"]:
+            if "type" in sever_list and sever_list["type"] == "CVSS_V3":
+                severity.append(
+                    VulnerabilitySeverity(
+                        system=SCORING_SYSTEMS["cvssv3_vector"],
+                        value=sever_list["score"],
+                    )
+                )
+    elif "ecosystem_specific" in raw_data and "severity" in raw_data["ecosystem_specific"]:
+        severity.append(
+            VulnerabilitySeverity(
+                system=SCORING_SYSTEMS["generic_textual"],
+                value=raw_data["ecosystem_specific"]["severity"],
+            )
+        )
+    else:
+        severity = []
+        logger.warning(f"severity not found- {raw_data['id']!r}")
+
+    references = []
+    if "references" in raw_data:
+        references = [
+            Reference(url=ref["url"], severities=severity) for ref in raw_data["references"] if ref
+        ]
+
+    affected_package = []
+    if "affected" in raw_data:
+        for affected_pkg in raw_data["affected"]:
+            package = affected_pkg["package"]
+            if "ecosystem" in package and "name" in package:
+                purl = PackageURL(type=package["ecosystem"], name=package["name"])
+            else:
+                purl = ""
+                logger.error(f"purl affected_pkg not found - {raw_data['id'] !r}")
+
+            affected_version_range = None
+            if "versions" in affected_pkg:
+                affected_version_range = (
+                    # HINT:  Values larger than 1/3 of a buffer page cannot be indexed.
+                    PypiVersionRange(affected_pkg["versions"][slice(200)])
+                )
+
+            if "ranges" in affected_pkg:
+                for fix_range in affected_pkg["ranges"]:
+                    fixed_version = set()
+                    if "type" in fix_range:
+                        if fix_range["type"] == "ECOSYSTEM":
+                            filter_fixed = list(
+                                filter(lambda x: x.keys() == {"fixed"}, fix_range["events"])
+                            )
+                            list_fixed = [i["fixed"] for i in filter_fixed]
+                            for i in list_fixed:
+                                try:
+                                    fixed_version.add(PypiVersion(i))
+                                except InvalidVersion:
+                                    logger.error(
+                                        f"fixed_version InvalidVersion - PypiVersion - {raw_data['id'] !r}"
+                                    )
+
+                        if fix_range["type"] == "SEMVER":
```

Comment:
How would it be possible for a PyPI range to use semver?
Also this code is too complicated for me to understand as is the same code above... can you find a way to decompose this in something simpler? Also it is duplicated almost exactly with the code above... so creating a function would make sense to avoid duplicated code.

Reply:
ex: `GHSA-2ghc-6v89-pw9j.json ` 
https://osv.dev/vulnerability/GHSA-2ghc-6v89-pw9j

Reply:
@ziadhany Thanks ... so in https://osv.dev/vulnerability/GHSA-2ghc-6v89-pw9j I see  three different affected packages:

- `npm/body-parser-xml` which would become the purl `pkg:npm/body-parser-xml` and is reported as SEMVER versioning which would become the `npm` vers scheme on our side. This is found at https://www.npmjs.com/package/body-parser-xml

- `NuGet/body-parser-xml-nuget` which would become the purl `pkg:nuget/body-parser-xml-nuget` and is reported as ECOSYSTEM versioning and would become the `nuget` vers scheme on our side... BUT this does NOT exist as a nuget package at all. https://www.nuget.org/packages?q=body-parser-xml ... not even close

- `PyPI/body-parser-xml-pip` which would become the purl `pkg:nuget/body-parser-xml-pip` and is reported as ECOSYSTEM versioning and would become the `pypi` vers scheme on our side. BUT this does NOT exist as a pypi package at all: https://pypi.org/search/?q=body-parser-xml


So what to make of this? OSV and GitHub data are likely not really reviewed after all. That's not the first time I see junk ghost records like this that are supposedly reviewed.

In anycase this are not a Pypi package... but it is going to be hard to figure this out ... at least not at import time.
I note that this is not seen at all in https://github.com/pypa/advisory-database 
So practically:
- I entered https://github.com/nexB/vulnerablecode/issues/644 to add improvers to validate purls
- short term we could flag these with lower confidence but I am not sure we have enough at import time
- so we can import these ... each should use their own specific vers scheme though and PyPI is NOT semver alright

There is not much harm done to import non-existing purl BTW because these would never be found in the wild.